docs/manuals/gui/page_builder.md

```diff
@@ -0,0 +1,202 @@
+
+You can bind your application variables to a property value by setting the property to a string
+that contains a Python expression that depends on the variables.
+
+Here is how we would use a Python variable to hold the text handled in the `input` control we
+have used so far.<br/>
+The new code looks like this:
+```py
+first_name="John"
+
+with tgb.Page() as page:
+    tgb.html("p", "User information:")
+    tgb.input("{first_name}", label="First name")
+```
+
+And the result is identical to what was shown above.
+
+# Using blocks
+
+The Taipy GUI blocks can help organize the elements on the page.
+
+In the following code, we use the `layout^` block to organize the controls on the page: 
+```py
+first_name="John"
+last_name="Carpenter"
+age=43
+
+with tgb.Page() as page:
+    tgb.html("p", "User information:")
+    with tgb.layout("4 1"):
+        with tgb.part():
+            tgb.input("{first_name}", label="First name")
+            tgb.input("{last_name}", label="Last name")
+            tgb.input("{age}", label="Age")
+        tgb.button("Submit")
+```
+
+The `layout` block is defined as having two columns, where the first column is four times larger
+than the second one.
+
+Here is the resulting display:
+<figure>
+    <img src="../tgb-5-d.png" class="visible-dark" />
+    <img src="../tgb-5-l.png" class="visible-light"/>
+    <figcaption>Controls layout</figcaption>
+</figure>
+
+# Invoking callbacks
+
+Because you can set functions to callback properties, the binding to callback functions is more
+flexible than when you define page content using text.
+
+## Default callbacks
+
+Default callbacks are invoked if not explicitly assigned to callback properties.
+
+Consider the following script:
+```py
+from src.taipy.gui import Gui
+import src.taipy.gui.builder as tgb
+
+def on_action(state, id):
+    if id == "my_button":
+        # Do something...
+        pass
+
+with tgb.Page() as page:
+    tgb.button("Press me", id="my_button")
+
+Gui(page).run()
+```
+
+The `button^` does not define its *on_action* property: Taipy looks for an *on_action()* function
+in the code and invokes it when the button is pressed.
+
+## Named callbacks
+
+The name of the callback function can also be used as a callback property value.
+
+The code changes would be like this:
+```py
+def my_button_pressed(state, id):
+    # Do something...
+    pass
+
+with tgb.Page() as page:
+    tgb.button("Press me", on_action="my_button_pressed")
+```
+
+The `button^` does not define its *on_action* property: Taipy looks for a *on_action()* function
+in the code and invokes it when the button is pressed.
+
+## Functions as callbacks
+
+You can also use the Python function as a callback property value:
+```py
+def my_button_pressed(state, id):
+    # Do something...
+    pass
+
+with tgb.Page() as page:
+    tgb.button("Press me", on_action=my_button_pressed)
+```
+
+This would have the same behavior as in the case where you would have used the function name.
```

Comment:
We might want to address these points:
- Directly set a dictionary or array to properties, no need to map it through variable string 
```
tgb.chart(
            data="{csvdata}",
            x="Day",
            selected_color="green",
            y=["Daily hospital occupancy", "Daily hospital occupancy"],
            label=["Entity", "Code"],
            mode=[None, "markers"],
            color=[None, "red"],
            type=(None, "scatter"),
            xaxis=(None, "x2"),
            layout={"grid": {"rows": 1, "columns": 2, "subplots": [["xy", "x2y"]], "roworder": "bottom to top"}},
            on_range_change="range_change",
            width="100%",
            height="100%",
            selected="{selected_indices}",
        )
```
- a page can be defined without context manager by the add method which could be chained as it return itself
```
page = tgb.Page()
page.add(tgb.text("Hello")).add(tgb.text("World"))
```

Reply:
I addressed the second point.
The first one should in my opinion be an addendum to the chart control, where it makes a lot of sense.

Reply:
I agree! As long as the user know that it is an option we are good. We can also add it for selector as they can quickly defines lov.